waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/DefaultMetaStoreClientFactory.java

```diff
@@ -155,9 +226,17 @@ CloseableThriftHiveMetastoreIface newInstance(
       int reconnectionRetries,
       ThriftMetastoreClientManager base) {
     ReconnectingMetastoreClientInvocationHandler reconnectingHandler = new ReconnectingMetastoreClientInvocationHandler(
-        name, reconnectionRetries, base);
-    return (CloseableThriftHiveMetastoreIface) Proxy
-        .newProxyInstance(getClass().getClassLoader(), INTERFACES, reconnectingHandler);
+            name, reconnectionRetries, base);
+    if (base.isSaslEnabled()) {
```

Comment:
we have a test `DefaultMetastoreClientFactory.closeNullBase` which test that base == null, this test I think can be deleted we don't check for null anywhere except in the close method.